backend/ml/tests/test_mehestan_individual.py

```diff
@@ -0,0 +1,138 @@
+            ("C", "D", random.randint(-10, 10)),
+        ]
+        comparisons_df = pd.DataFrame(comparisons, columns=["entity_a", "entity_b", "score"])
+        scores = compute_method(comparisons_df)
+        assert scores.raw_score.mean() == pytest.approx(1.0, abs=1e-4)
+
+    def test_comparisons_strong_preferences(self, compute_method):
+        if compute_method is compute_individual_score:
+            pytest.xfail("The non-BBT version does not preserve preferences order :( ")
+
+        comparisons = [
+            ("A", "B", 0),
+            ("A", "C", 0),
+            ("A", "D", 0),
+            ("B", "C", 0),
+            ("B", "D", 0),
+            ("C", "D", 0),
+            # strong preferences
+            ("A", "E", 9),
+            ("B", "F", 10),
+        ]
+        comparisons_df = pd.DataFrame(comparisons, columns=["entity_a", "entity_b", "score"])
+        scores = compute_method(comparisons_df)
+
+        # F is preferred to E
+        assert scores.loc["F"].raw_score > scores.loc["E"].raw_score
+
+    def test_comparisons_monotony(self, compute_method):
+        if compute_method is compute_individual_score:
+            pytest.xfail("The non-BBT version does not preserve monotony")
+
+        comparisons_1 = [
+            ("A", "B", 8),
+            ("A", "C", 9),
+            ("A", "D", 10),
+            ("B", "C", 8),
+            ("B", "D", 9),
+            ("C", "D", 10),
+        ]
+
+        # Move preferences towards A
+        comparisons_2 = [
+            ("A", "B", 7),
+            ("A", "C", 8),
+            ("A", "D", 9),
+            ("B", "C", 8),
+            ("B", "D", 9),
+            ("C", "D", 10),
+        ]
+
+        comparisons_1_df = pd.DataFrame(comparisons_1, columns=["entity_a", "entity_b", "score"])
+        scores_1 = compute_method(comparisons_1_df)
+        comparisons_2_df = pd.DataFrame(comparisons_2, columns=["entity_a", "entity_b", "score"])
+        scores_2 = compute_method(comparisons_2_df)
+
+        assert scores_2.loc["A"].raw_score > scores_1.loc["A"].raw_score
+
+    def test_comparisons_non_connex(self, compute_method):
+        comparisons = [
+            # Group 1
+            ("A", "B", -1),
+            ("A", "C", 0),
+            ("A", "D", 1),
+            # Group 2
+            ("E", "F", -1),
+            ("E", "G", 0),
+            ("E", "H", 1),
+        ]
+        comparisons_df = pd.DataFrame(comparisons, columns=["entity_a", "entity_b", "score"])
+        scores = compute_method(comparisons_df)
+
+        assert scores.loc["A"].raw_score == pytest.approx(0.0, abs=1e-4)
+        assert scores.loc["A"].raw_score == pytest.approx(scores.loc["E"].raw_score, abs=1e-4)
+        assert scores.loc["B"].raw_score == pytest.approx(scores.loc["F"].raw_score, abs=1e-4)
+        assert scores.loc["C"].raw_score == pytest.approx(scores.loc["G"].raw_score, abs=1e-4)
+        assert scores.loc["D"].raw_score == pytest.approx(scores.loc["H"].raw_score, abs=1e-4)
```

Comment:
I've read this sentence in the paper that with CBT : 

> Essentially, we consider that when a
> contributor reports rab, it is as though answered they preferred a a (non-integer) number 1 + rab
> of times, and they reported preferring b a number 1 − rab of times

Shouldn't we have a test for this ?
Like, have 3 videos A, B, C and D with the same rating and set the comparisons :
("A", "B", 0)
("A", "C", 0)
("A", "D", 0)
("B", "C", 0)
("B", "D", 0)
("C", "D", 0)
("A", "E", 2)
("B", "E", 8)
("C", "F", 5)
("D", "F", 5)
assert score.loc["E"].raw_score == pytest.approx(scores.loc["F"].raw_score, abs=1e-4)

Reply:
It almost works but does not quite. It took us a bit of time to figure this one out! 
It would work perfectly if the scores of A, B, C and D were equal at optimum.
However, they will themselves be moved by different amounts towards negative scores.